src/energyhub.py

```diff
@@ -874,222 +1004,135 @@ def _monte_carlo_networks(self, netw):
-                    return (
-                        b_arc.var_capex_aux
-                        == b_arc.var_size * b_netw.para_capex_gamma1
-                        + b_netw.para_capex_gamma2
-                    )
-                elif economics.capex_model == 2:
-                    return (
-                        b_arc.var_capex_aux
-                        == b_arc.var_size * b_arc.distance * b_netw.para_capex_gamma1
-                        + b_netw.para_capex_gamma2
-                    )
-                elif economics.capex_model == 3:
-                    return (
-                        b_arc.var_capex_aux
-                        == b_arc.var_size * b_arc.distance * b_netw.para_capex_gamma1
-                        + b_arc.var_size * b_netw.para_capex_gamma2
-                        + b_netw.para_capex_gamma3
-                    )
+            b_arc = netw_data._define_capex_arc(b_arc, b_netw, arc[0], arc[1])
 
-            b_arc.const_capex_aux = Constraint(rule=init_capex)
-            self.solver.add_constraint(b_arc.const_capex_aux)
+            if b_arc.big_m_transformation_required:
+                b_arc = perform_disjunct_relaxation(b_arc)
 
-    def _monte_carlo_import_prices(self, node, car):
+    def _monte_carlo_import_prices(self, period, node, car, MC_import_row=None):
         """
         Changes the import prices
         """
         config = self.data.model_config
+        aggregation_type = "full"
 
         sd = config["optimization"]["monte_carlo"]["sd"]["value"]
         sd_random = np.random.normal(1, sd)
 
-        model = self.model
-        set_t = model.set_t_full
-
-        import_prices = self.data.node_data[node].data["import_prices"][car]
-        b_node = model.node_blocks[node]
-
+        model = self.model[aggregation_type]
+        import_prices = self.data.time_series[aggregation_type][period][node][
+            "CarrierData"
+        ][car]["Import price"]
+        b_period_cost = model.block_costbalance[period]
+        # TODO: check for averaging
+        set_t = model.periods[period].set_t_full
+
+        # Update parameter
         for t in set_t:
-            # Update parameter
-            b_node.para_import_price[t, car] = import_prices.iloc[t - 1] * sd_random
+            if MC_import_row is not None:
+                model.periods[period].node_blocks[node].para_import_price[t, car] = (
+                    random.uniform(MC_import_row["min"], MC_import_row["max"])
+                )
+            else:
+                model.periods[period].node_blocks[node].para_import_price[t, car] = (
+                    import_prices.iloc[t - 1] * sd_random
+                )
 
-            # Remove constraint (from persistent solver and from model)
-            self.solver.remove_constraint(model.const_node_cost)
-            model.del_component(model.const_node_cost)
+        # delete old constraint
+        b_period_cost.del_component(b_period_cost.const_cost_import)
 
```

Comment:
This is a bit prone to problems: if we ever change the constraint below in the construct_balance, we will for sure forget it here. So my suggestion is: in construct_balances - construct_system_cost() we call private functions from the module construct_balances. 

So there is a bit of refactoring required:
in construct_system_cost we have:

b_period = _construct_capex_tecs(b_period)
b_period = _construct_capex_netw(b_period)
b_period = _construct_opex_tecs(b_period)
...

with  _construct_opex_tecs(b_period) and all the other functions defined there. in the energyhub we can then just call these functions, if required.

Reply:
@JeanWi